#165413097 Setup Api Documentation

Niyitangasam opened the pull request:
Description
=======
The ability of API is to describe its usage to the consumers. This PR set up API documentation using swagger and we will continue to update it throughout the development process.

Type of change
=======
N/A

How has it been tested
=======
It can be tested through  `/docs` appended on the base URL of the hosting server

Checklist:
=======
N/A

Pivotal tracker story ID
=======
[165413097](https://www.pivotaltracker.com/n/projects/2326984/stories/165413097)

AJAkimana reviewed (approved):
Great work @Niyitangasam.

Kabalisa commented:
Good Job @Niyitangasam. swagger is set up well all you can do is add endpoint documentations as we do them.

macphilips commented on index.js:
```diff
@@ -1,93 +1,53 @@
 
-require("./models/User");
+// route for swagger.json
+app.get('/swagger.json', function(req, res) {
```
what is the purpose of this route?

macphilips commented on package.json:
```diff
@@ -4,29 +4,34 @@
     "passport-local": "^1.0.0",
     "request": "^2.87.0",
-    "slug": "^0.9.1",
+    "save": "^2.3.3",
```
What are you using this package for?

macphilips commented on index.js:
```diff
@@ -1,93 +1,53 @@
 const app = express();
 
-app.use(cors());
+// swagger definition
```
Can you define the swagger configuration in a separate file and add it to a config folder

macphilips replied:
Also, add sample documentation, so your peers know how to you it

macphilips commented on index.js:
```diff
@@ -1,93 +1,53 @@
+// options for swagger jsdoc 
+const options = {
+    swaggerDefinition: swaggerDefinition,
+    apis: ['./routes/*.js']
```
update the pattern to recursive search any js file within the directory:
```suggestion
    apis: ['./routes/**/*.js']
```

macphilips commented on index.js:
```diff
@@ -1,93 +1,53 @@
+      version: '1.0.0',
+      description: 'A Social platform for the creative at heart',
+    },
+    host: '',
```
You need to make the host field configurable by using environment variables. You can do this by using [dotenv](https://www.npmjs.com/package/dotenv) package.
```suggestion
    host: `${process.env.BASE_URL}`,
```

macphilips commented on index.js:
```diff
@@ -1,93 +1,53 @@
+      description: 'A Social platform for the creative at heart',
+    },
+    host: '',
+    basePath: '/',
```
```suggestion
    basePath: '/api/v1',
```

macphilips reviewed (changes requested):
You've done a nice job so far, but this task is not complete. I have made some comments kindly attend to it.

ericnyirimana committed:
chore(docs): setup API documentation using swagger

- add swagger js doc in package.json
- add swagger UI in package.json
- define `/docs` as endpoint fro documentation
- Add swagger js file with login in config folder`
- Define model and endpoint functionalities of login  in routes

[Finishes #165413097]
